src/utils.js

```diff
@@ -1,241 +1,227 @@
-		}
-		copy[key] = exports.clone(obj[key]);
-	}
-	return copy;
-};
-
-/**
-int nIndexOf (string, string match, int index)
-
-Finds a match in a string at a given index
-
-Usage:
-string = "My name is Rive"
-match = " "
-index = 2
-return = 7
-
-Summary: It will look for a second space in the string
-*/
-exports.nIndexOf = function(string, match, index) {
-	return string.split(match, index).join(match).length;
-};
+ * Object clone (object)
+ *
+ * Clone an object.
+ */
+
+const clone = (obj) => {
+    if (obj === null || typeof obj !== "object") {
+      return obj
+    }
+    const copy = obj.constructor()
+    for (const key in obj) {
+      if (!obj.hasOwnProperty(key)) {
+        copy[key] = clone(obj[key])
+      }
+    }
+    return copy
+  },
+
+  /**
+   * Void extend (object a, object b)
+   *
+   * Combine the properties of both objects into one. The properties from
+   * object 'b' are inserted into 'a'.
+   */
+
+  extend = (a, b) => (() => {
+    const result = []
+    for (const attr in b) {
+      const value = b[attr]
+      if (!b.hasOwnProperty.call(attr)) {
+        result.push(a[attr] = value)
+      }
+    }
+    return result
+  })(),
+
+  /**
+   * Boolean isAPromise (object)
+   *
+   * Determines if obj looks like a promise
+   */
+
+  isAPromise = (obj) => obj &&
+  obj.then &&
+  obj.catch &&
+  obj.finally &&
+  typeof obj.then === "function" &&
+  typeof obj.catch === "function" &&
+  typeof obj.finally === "function",
+
+  /**
+   * Bool isAtomic (string trigger)
+   *
+   * Determine whether a trigger is atomic.
+   */
+
+  isAtomic = (trigger) => {
+    /**
+     * Atomic triggers don't contain any wildcards or parenthesis or
+     * anything of the sort. We don't need to test the full character set
+     * just left brackets will do.
+     */
+    for (const special of[
+        "*",
+        "#",
+        "_",
+        "(",
+        "[",
+        "<",
+        "@"
+      ]) {
+      if (trigger.includes(special)) {
+        return false
+      }
+    }
+    return true
+  },
+
+  /**
+   * Int nIndexOf (string, string match, int index)
+   *
+   * Finds a match in a string at a given index
+   *
+   * Usage:
+   * string = "My name is Rive"
+   * match = " "
+   * index = 2
+   * return = 7
+   *
+   * Summary: It will look for a second space in the string
+   */
+
+  nIndexOf = (string, match, index) => string.
+split(match, index).
+join(match).length,
+
+  /**
+   * String quotemeta (string)
+   *
+   * Escape a string for a regexp.
+   */
+
+  quoteMeta = (string) => {
+    const unsafe = "\\.+*?[^]$(){}=!<>|:".split("");
+    for (const char of unsafe) {
+      string = string.replace(new RegExp(`\\${char}`, "g"), `\\${char}`)
+    }
+    return string
+  },
+
+  /**
+   * String stringFormat (string type, string)
+   *
+   * Formats a string according to one of the following types:
+   * - formal
+   * - sentence
+   * - uppercase
+   * - lowercase
+   */
+
+  stringFormat = (type, string, first) => {
```

Comment:
you've changed the method signature here, adding a new parameter `first`, whose value is never used (it's only ever re-assigned); may want to declare it with `let` at the top of the block instead (and maybe consider using a `switch` statement here for readability?)

Reply:
I think it will be better, good idea.